src/main/webapp/app/shared/links/TextWithRefs.tsx

```diff
@@ -0,0 +1,47 @@
+import { REFERENCE_LINK } from 'app/config/constants/constants';
+import { ParsedRef, parseReferences } from 'app/oncokb-commons/components/RefComponent';
+import React from 'react';
+
+export interface ITexWithRefsProps {
+  content: string;
+}
+
+/* eslint-disable @typescript-eslint/prefer-regexp-exec */
+export default function TextWithRefs({ content }: ITexWithRefsProps) {
+  const result: JSX.Element[] = [];
+  const parts = content?.split(REFERENCE_LINK) || [];
+
+  function formatReferences(refs: ParsedRef[]) {
+    const prefix = refs[0].prefix;
+    const joinedReferences = refs.map((ref, index) => {
+      return (
+        <span key={index}>
+          <a target="_blank" rel="noopener noreferrer" href={ref.link}>
+            {`${prefix}${ref.content}`}
+          </a>
+          {index !== refs.length - 1 ? ', ' : null}
+        </span>
+      );
+    });
+
+    return (
+      <span>
+        {'('}
+        {joinedReferences}
+        {')'}
+      </span>
+    );
+  }
+
+  for (let i = 0; i < parts.length; i++) {
+    const part = parts[i];
+    if (part.match(REFERENCE_LINK)) {
```

Comment:
Will this ever be true? When you split the content using `REFERENCE_LINK`, wouldn't all ref been removed from parts?

Reply:
It will, since the abstract title is included in the capture group, it will be included in the output of `.split()`

Reply:
The split will remove the abstract capture group though, would it?

Reply:
It will actually keep the abstract capture group. Here is an example: https://onecompiler.com/javascript/425tvvb94

Reply:
Oh interesting. So if you have the regex capture group, it will remain in the list. https://onecompiler.com/javascript/425wyh6mq if you remove the capture group, it will remove the matched string.